.github/workflows/build-publish-develop-pr.yml

```diff
@@ -0,0 +1,98 @@
+name: "Build and Publish Chainlink"
+
+on:
+  pull_request:
+    types:
+      - labeled
+  push:
+    branches:
+      - develop
+      - "release/**"
+
+jobs:
+  goreleaser-build-publish-chainlink:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+        goarch: [amd64, arm64]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+
+      - name: Get image tag
+        id: get-image-tag
+        run: |
+          short_sha=$(git rev-parse --short HEAD)
+          echo "build-publish=false" | tee -a $GITHUB_OUTPUT
+          if [[ ${{ github.event_name }} == 'push' ]]; then
+            if [[ ${{ github.ref_name }} == 'release/'* ]]; then
+              echo "image-tag=release-${short_sha}" | tee -a $GITHUB_OUTPUT
+              echo "build-publish=true" | tee -a $GITHUB_OUTPUT
+            else
+              echo "image-tag=develop-${short_sha}" | tee -a $GITHUB_OUTPUT
+              echo "build-publish=true" | tee -a $GITHUB_OUTPUT
+            fi
+          else
+            if [[ ${{ github.event.label.name }} == 'build-publish' ]]; then
+              echo "image-tag=pr-${{ github.event.number }}-${short_sha}" | tee -a $GITHUB_OUTPUT
+              echo "build-publish=true" | tee -a $GITHUB_OUTPUT
+            fi
+          fi
+
+      - name: Configure aws credentials
+        if: steps.get-image-tag.outputs.build-publish == 'true'
+        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+        with:
+          role-to-assume: ${{ secrets.AWS_OIDC_IAM_ROLE_BUILD_PUBLISH_DEVELOP_PR }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          mask-aws-account-id: true
+          role-session-name: goreleaser-build-publish-chainlink
+
+      - name: Build and publish images
+        if: steps.get-image-tag.outputs.build-publish == 'true'
+        uses: ./.github/actions/goreleaser-build-sign-publish
+        with:
+          enable-docker-publish: "true"
```

Comment:
Did we want this to be false unless it was from a PR with a label?

Reply:
we are enabling to publish to the private sdlc ecr. it's a custom thing we created because the regular goreleaser release with snapshot does not do publish at all.

or did you mean we don't want to publish for any pr - just pr with the label `build-publish`?

https://github.com/smartcontractkit/chainlink/blob/96829b0a3afd745e928191a4786503fa65734a0e/.github/actions/goreleaser-build-sign-publish/action_utils#L83-L86

Reply:
> did you mean we don't want to publish for any pr - just pr with the label build-publish

This. Just to minimize the number of images we publish that won't ever get used. But I'm assuming we would use this workflow to build and then optionally publish. Is there still a separate workflow that builds?

Reply:
ahh I had created the conditional here:
https://github.com/smartcontractkit/chainlink/blob/96829b0a3afd745e928191a4786503fa65734a0e/.github/workflows/build-publish-develop-pr.yml#L38-L43
so only the label `build-publish` will be built and published

Reply:
I see. That will skip the build and publish docker job if the label is not on the PR, right? I guess I was assuming that we would have this workflow to cover all normal builds too (PR's without the label) and in those cases, we don't want to publish the image (if the label isn't on the PR).

Reply:
I'd like it if we could hook our E2E tests (which run on almost every PR) into this build step as well and not have our own custom one any more. Is having the tag necessary for this?

Reply:
yeah great point - looking over your ticket I had added this :)
https://github.com/smartcontractkit/chainlink/blob/37d617bfc17f375095e99f578728814cb9f588b5/.github/workflows/build-publish-develop-pr.yml#L9-L17
would this solve your issue?
of course we will need to tweak the `goreleaser-build-sign-publish` composite action a bit

Reply:
@kalverra that's going to be big (avoid all of those unnecessary builds). Glad you're thinking about that.

We might need to push all images from PRs to ECR though if you're going to use it from tests (unless we used artifacts but...). In that case maybe we have a way to differentiate between docker images from PR's that can be pruned out quickly via lifecycle rules (only used for tests) and docker images from PR's that might need to run in longer tests.

Reply:
Right now it does, but I think @kalverra is considering using this image for the tests. If so, we'd need to push all PR images to ECR for your tests, right?